packages/backend/src/open_payments/auth/middleware.ts

```diff
@@ -145,19 +151,16 @@ export function createTokenIntrospectionMiddleware({
               : undefined
         }
       }
+
+      await next()
```

Comment:
Let's keep this in the outside of try-catch. Otherwise, the catch block in this middleware will apply to all of the following middlewares called in `await next()` which might lead to unexpected behaviour, e.g., adding `WWW-Authenticate` when we don't want to.